### ISSUE TITLE
Add project: search project by URL

### ISSUE THREAD

**Issue by humitos** [readthedocs/ext-theme#451]
It would be good to be able to just paste the GitHub URL of the project and get the exact result.

![Peek 2024-08-14 17-13](https://github.com/user-attachments/assets/c4fd2ad7-a633-446d-93c5-47fbe8d559dc)

That will be a lot quicker. Besides, it would avoid typos, or even importing repositories with the same name (forks from another organization/user).

<img src="https://front.com/assets/img/favicons/favicon-32x32.png" height="16" width="16" alt="Front logo" /> [Front conversations](https://app.frontapp.com/open/top_11uvwn)

**Comment by humitos** [comment]
It seems the filtering is done in the backend when hitting the API endpoint: https://github.com/readthedocs/ext-theme/blob/b1ad973e2693ace11f43ec9d1b808b28b0c22b21/src/js/project/create.js#L130

So, if we want to go that way, we would need to add an extra filter in the endpoint: https://docs.readthedocs.io/en/stable/api/v3.html#get--api-v3-remote-repositories-

**Comment by humitos** [comment]
We can probably pass `&html_url=` instead of `full_name` since it will work for both since the full name is contained on the `html_url` field as well.

**Comment by agjohnson** [comment]
Not necessarily, some users rename their projects to verbose names that don't match their repo name. The change here is likely altering the API to be able to search both/multiple fields.

There are other Project fields that might be useful to query as well. I'm likely describing `?expand=projects&search={query}` -- or is there a native DRF version of this?

This should probably be moved to the application. The front end would just use a different field name in the query.

It's also a good issue for later work on new features, I'll put this on the new features milestone, it's not a high priority/bug.
